include/deal.II/distributed/solution_transfer.h

```diff
@@ -276,16 +269,17 @@ namespace parallel
         const typename Triangulation<dim, DoFHandlerType::space_dimension>::
           cell_iterator &cell,
         const typename Triangulation<dim, DoFHandlerType::space_dimension>::
-          CellStatus               status,
-        const void *               data,
+          CellStatus status,
+        const boost::iterator_range<std::vector<char>::const_iterator>
+                                   data_range,
         std::vector<VectorType *> &all_out);
 
 
       /**
        *
        */
       void
-      register_data_attach(const std::size_t size);
+      register_data_attach();
```

Comment:
This is not your fault, but while you're there, would you mind writing a comment what this function does? A couple of sentences is good enough.